src/pytest_plugins/test_filler/test_filler.py

```diff
@@ -381,6 +407,37 @@ def t8n(request, evm_bin: Path) -> Generator[TransitionTool, None, None]:
     t8n.shutdown()
 
 
+@pytest.fixture(autouse=True, scope="session")
+def alloc_class(request) -> Type[Alloc]:
+    """
+    Modifies the `Alloc` class for all instances to have the same configuration.
+    """
+    kw_args: Dict[str, Any] = {}
+    if request.config.getoption("strict_alloc"):
+        kw_args["alloc_mode"] = AllocMode.STRICT
+    if request.config.getoption("test_contract_start_address"):
+        kw_args["start_contract_address"] = int(
+            request.config.getoption("test_contract_start_address"), 0
+        )
+    if request.config.getoption("test_contract_address_increments"):
+        kw_args["contract_address_increments"] = int(
+            request.config.getoption("test_contract_address_increments"), 0
+        )
+
+    class AllocSubclass(Alloc, **kw_args):
+        pass
+
+    return AllocSubclass
+
+
+@pytest.fixture(autouse=True)
+def pre(alloc_class: Type[Alloc]) -> Alloc:
+    """
+    Returns the default pre allocation for all tests (Empty alloc).
+    """
+    return alloc_class()
```

Comment:
While coverting the Shanghai tests it got me thinking that we should autouse `post` and `env` fixtures too:
```python
@pytest.fixture
def env() -> Environment:
    return Environment()

@pytest.fixture
def post() -> Alloc:
    return Alloc()
```

Maybe the `sender` aswell?
```python
@pytest.fixture
def sender(pre: Alloc) -> EOA:
    return pre.fund_eoa(1_000_000_000_000_000)
```

Being able to immediately use these defaults within a test function would be really nice:
```python
def test_warm_coinbase_gas_usage(
    state_test: StateTestFiller,
    env: Environment,
    pre: Alloc,
    sender: Address,
    ...
):
...
```

Reply:
This is really nice, even if the user does not use them, it would be nice to have the option of adding a line a pre-setup with defaults.

Reply:
Once I get back to the EEST utils PR, like you mentioned there, we could have a command that automatically creates a template test and then utilze this within the template etc

Reply:
I tried to do this and with the new parameters there were more imports to do every time and it ended up looking not so better. So I opted for specifying a default value for `fund_eoa` and we can revisit this idea if we find a more compelling reason to actually do it.